content-api/content-service/app/controllers/BaseController.scala

```diff
@@ -68,16 +68,19 @@ abstract class BaseController(protected val cc: ControllerComponents)(implicit e
         })
     }
 
-    def getRequest(input: java.util.Map[String, AnyRef], context: java.util.Map[String, AnyRef], operation: String): org.sunbird.common.dto.Request = {
+    def getRequest(input: java.util.Map[String, AnyRef], context: java.util.Map[String, AnyRef], operation: String, categoryMapping: Boolean = false): org.sunbird.common.dto.Request = {
+        //Todo mapping and reverse mapping
+        if (categoryMapping) setContentAndCategoryTypes(input)
         new org.sunbird.common.dto.Request(context, input, operation, null);
     }
 
-    def getResult(apiId: String, actor: ActorRef, request: org.sunbird.common.dto.Request) : Future[Result] = {
+    def getResult(apiId: String, actor: ActorRef, request: org.sunbird.common.dto.Request, categoryMapping: Boolean = false) : Future[Result] = {
         val future = Patterns.ask(actor, request, 30000) recoverWith {case e: Exception => Future(ResponseHandler.getErrorResponse(e))}
         future.map(f => {
             val result = f.asInstanceOf[Response]
             result.setId(apiId)
             setResponseEnvelope(result)
+            if (categoryMapping) setContentAndCategoryTypes(result.getResult.getOrDefault("content", new util.HashMap[String, AnyRef]()).asInstanceOf[util.Map[String, AnyRef]])
```

Comment:
Can we do this for ok response only, not for all the responses.

Reply:
Done